src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,62 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int ARRAY_MAX_SIZE = 10;
+    private Pair<K, V>[] storage;
+    private int indexOfNextPair = 0;
+
+    public class Pair<K, V> {
+        private K key;
+        private V value;
+
+        public Pair(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+    }
+
+    public StorageImpl() {
+        storage = new Pair[ARRAY_MAX_SIZE];
```

Comment:
you need to cast your array
```suggestion
        storage = (Pair<K, V>[]) new Pair[ARRAY_MAX_SIZE];
```